controller/appcontroller.go

```diff
@@ -1195,6 +1195,14 @@ func (ctrl *ApplicationController) processRequestedAppOperation(app *appv1.Appli
 				ctrl.requestAppRefresh(app.QualifiedName(), CompareWithLatest.Pointer(), &retryAfter)
 				return
 			} else {
+				if state.Phase == synccommon.OperationRunning && app.Spec.SyncPolicy.Retry.Refresh && app.Status.Sync.Revision != state.Operation.Sync.Revision {
+					logCtx.Infof("A new revision is available, refreshing and terminating app, was phase: %s, message: %s", state.Phase, state.Message)
+					ctrl.requestAppRefresh(app.QualifiedName(), CompareWithLatest.Pointer(), nil)
```

Comment:
This forces a refresh, without it, it waits for the next window, which can take some time.